chainforge/react-server/src/store.js

```diff
@@ -238,8 +238,16 @@ const useStore = create((set, get) => ({
 
     // Pull data from each source recursively:
     const pulled_data = {};
+    let var_history = new Set();
     const get_outputs = (varnames, nodeId) => {
       varnames.forEach(varname => {
+        // Check for duplicate variable names
+        if (var_history.has(String(varname).toLowerCase())) {
+          throw "Duplicate variable name, {" + varname + "}, in the same chain is not allowed."
```

Comment:
Can you create a new Error type for this? You might create an errors.ts file in the backend/ folder, declare it and import it from there. 

See https://engineering.udacity.com/handling-errors-like-a-pro-in-typescript-d7a314ad4991 for help with this. You can then throw that specific error, and that error object can detected in the `catch` branch and specifically caught.